src/ir/gc-type-utils.h

```diff
@@ -117,8 +117,12 @@ inline EvaluationResult evaluateKindCheck(Expression* curr) {
       case RefAsI31:
         expected = I31;
         break;
+      case ExternInternalize:
+      case ExternExternalize:
+        // These instructions can never be removed.
+        return Unknown;
       default:
-        WASM_UNREACHABLE("unhandled BrOn");
+        WASM_UNREACHABLE("unhandled RefAs");
```

Comment:
Let's get rid of the `default` case here so the compiler will flag this next time we add a new op.

Reply:
Yeah, that might make sense. But then the compiler can't tell that `expected` is always set in all control flow paths - it doesn't know we treat warnings as errors. I can avoid that warning by setting a default that isn't needed, but that's a little less clean, requires a comment, and also may be slower to execute (the optimizer can't remove it; though that's probably trivial). Overall, I'm not sure which is better.

Reply:
Which way do you prefer @tlively ?

Reply:
Keeping the unreachable default seems fine. I think we do that in many places.